test/common/filesystem/filesystem_impl_test.cc

```diff
@@ -121,6 +121,49 @@ TEST_F(FileSystemImplTest, CanonicalPathFail) {
 }
 #endif
 
+TEST_F(FileSystemImplTest, splitFileName) {
+  std::string name;
+  name.reserve(255);
+  std::string path = "/foo/bar/baz";
+  file_system_.splitFileName(path, name);
+  EXPECT_EQ(path, "/foo/bar");
+  EXPECT_EQ(name, "baz");
+  file_system_.splitFileName(path, name);
+  EXPECT_EQ(path, "/foo");
+  EXPECT_EQ(name, "bar");
+  file_system_.splitFileName(path, name);
+  EXPECT_EQ(path, "/");
+  EXPECT_EQ(name, "foo");
+  file_system_.splitFileName(path, name);
+  EXPECT_EQ(path, "/");
+  EXPECT_EQ(name, "");
+  std::string invalid_path = "nopathdelimeter";
+  EXPECT_THROW(file_system_.splitFileName(invalid_path, name), EnvoyException);
+#ifdef WIN32
```

Comment:
Please split this test into platform variants in different test files, and include them using bazel, same as the prod code. Same elsewhere.

Reply:
Question on this approach; how will maintainers ensure that unix/bash_impl_test.cc evaluations are also updated with win32/bash_impl_test.cc?

I could see bash_impl_test.cc setting out the common case, with specific bash_impl_test_win32.cc or bash_impl_test_linux.cc cases which apply only on a specific platform. WDYT?

Reply:
> I could see bash_impl_test.cc setting out the common case, with specific bash_impl_test_win32.cc or bash_impl_test_linux.cc cases which apply only on a specific platform. WDYT?

Yes, +1

Reply:
@mattklein123 we agree with splitting out this issue, already present across several files in /test/, and are happy to sign up for that work. Refactoring out these preexisting WIN32 blocks can be revisited in a subsequent porting round. It should not block this PR IMO, and introduces more time for this port to further drift from master. Can we stay focused on the proposed scope for this PR?

Reply:
Yes that's fine.